File: gym-backend/src/main/java/com/in/GymManagementSystem/controller/AttendanceService.js
import axios from "axios";

const API_URL = "http://localhost:9999/api/attendance";

export const getAllAttendance = async () => {
  const response = await axios.get(API_URL);
  return response.data;
};

export const getAttendanceByDate = async (date) => {
  const response = await axios.get(`${API_URL}/date/${date}`);
  return response.data;
};

export const getAttendanceByMember = async (memberId) => {
  const response = await axios.get(`${API_URL}/member/${memberId}`);
  return response.data;
};

export const createAttendance = async (attendanceData) => {
  const response = await axios.post(API_URL, attendanceData);
  return response.data;
};

export const checkOut = async (id) => {
  const response = await axios.put(`${API_URL}/${id}/checkout`);
  return response.data;
};

File: gym-backend/src/main/java/com/in/GymManagementSystem/service/AttendanceService.java
package com.in.GymManagementSystem.service;

import com.in.GymManagementSystem.dto.AttendanceDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import java.time.LocalDate;
import java.util.List;

public interface AttendanceService {
    List<AttendanceDTO> getAllAttendance();

    Page<AttendanceDTO> getAllAttendancePaged(Pageable pageable);

    List<AttendanceDTO> getAttendanceByDate(LocalDate date);

    List<AttendanceDTO> getAttendanceByMember(Long memberId);

    AttendanceDTO createAttendance(AttendanceDTO attendanceDTO);

    AttendanceDTO checkOut(Long id);

    void deleteAttendance(Long id);
}
